Feature/geo

JoshuaWachana opened the pull request:
### Changes

I made a lot of changes/additions to files related to the Geo-Demographics feature. Most of the important work was done in GeoMaps.jsx, USMap.jsx and WorldMaps.jsx. The later two are files that I created in this PR. Additionally, I created a CSS file GeoMap.css specifically to add styles related to this feature because indec.css was getting to clunky. 

### Feedback being sought

Do you see any ways the code related to this feature could be improved? I would also like a bit of feedback in regards to the styling of this feature. Does it match the overall styling of the project, or should I make some amendments?

harshita-jaiswal commented on NBA-Stats-Dashboard/src/components/GeoMaps.jsx:
```diff
@@ -1,10 +1,73 @@
+      <div className='mapSection mapSection3'>
+        <fieldset className='mapSelectionField'>
+          <legend>MAP Selection</legend>
+          <label>
```
Please use explicit labeling. Also, the 'for' attribute of the label should be associated with the input's id being labeled.
Refer: https://www.w3.org/WAI/tutorials/forms/labels/#:~:text=Associating%20labels%20implicitly,-In%20some%20situations&text=In%20this%20case%2C%20the%20label,better%20supported%20by%20assistive%20technology.

JoshuaWachana replied:
Fixed! I used the htmlFor attribute and made sure it matches the id of the corresponding radio button input.

harshita-jaiswal commented on NBA-Stats-Dashboard/src/components/GeoMaps.jsx:
```diff
@@ -1,10 +1,73 @@
+            />{' '}
+            US Map
+          </label>
+          <label>
```
Same as above

JoshuaWachana replied:
Fixed! Like the other situation, I used the htmlFor attribute and made sure it matches the id of the corresponding radio button input.

harshita-jaiswal commented on NBA-Stats-Dashboard/src/components/GeoMaps.jsx:
```diff
@@ -1,10 +1,73 @@
+              name='map'
+              value='us'
+              checked={mapSelected.us}
+              onChange={mapSelection}
```
Could you explain why the onChange event was chosen over the onClick event, and what the benefits and differences are? Also, is there an equivalent to onChecked for radio buttons?

JoshuaWachana replied:
Hello Harshita, I used the onChange event first and it seemed to work well, so I just stuck with it. Using onClick would also be another possibility, but I do not see any reason to change it at this stage since the former seems to be doing it's job well. Unfortunately, there is no onChecked event, that is something the developers should look into. However, there is a checked property for radio buttons which I utilized a bit in this PR.

harshita-jaiswal commented:
It's good to keep the page-specific CSS in different CSS files. Even I was planning to do the same but it's good you started. Overall, I like the design, and it's good to go for all the pages.

JoshuaWachana commented:
> It's good to keep the page-specific CSS in different CSS files. Even I was planning to do the same but it's good you started. Overall, I like the design, and it's good to go for all the pages.

Awesome. I have addressed your comments/feedback and updated the PR with a new commit. Thanks for the feedback.
